src/main/java/cz/mzk/fedora/documentcorrection/RestClient.java

```diff
@@ -0,0 +1,117 @@
+package cz.mzk.fedora.documentcorrection;
+
+import org.apache.tomcat.util.codec.binary.Base64;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+import org.w3c.dom.*;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.*;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.Objects;
+
+public class RestClient {
+
+    private final String fedoraHost;
+    private final HttpEntity<String> httpEntity;
+    private final RestTemplate restTemplate;
+    private final DocumentBuilder xmlParser;
+    private final Transformer xmlTransformer;
+    private final XPathExpression xmlPathExp;
+
+    public RestClient(String fh, String fu, String fp)
+            throws ParserConfigurationException, XPathExpressionException, TransformerException {
+        fedoraHost = fh;
+        restTemplate = new RestTemplate();
+        HttpHeaders httpHeaders = createHttpHeaders(fu, fp);
+        httpEntity = new HttpEntity<>(httpHeaders);
+
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newDefaultInstance();
+        factory.setNamespaceAware(true);
+        xmlParser = factory.newDocumentBuilder();
+
+        XPathFactory xFactory = XPathFactory.newInstance();
+        XPath xmlPath = xFactory.newXPath();
+        xmlPathExp = xmlPath.compile("//*[local-name() = 'isMemberOfCollection']");
+
+        TransformerFactory tFactory = TransformerFactory.newInstance();
+        xmlTransformer = tFactory.newTransformer();
+    }
+
+    public Document removeAllVC(Document doc, String vc) throws XPathExpressionException {
+        if (doc == null) { return null; }
+
+        String uuidVC = getFullFormatVC(vc);
+        NodeList nodes = (NodeList) xmlPathExp.evaluate(doc, XPathConstants.NODESET);
+        for (int i = 0; i < nodes.getLength(); i++) {
+            Node node = nodes.item(i);
+            if (node.getAttributes().item(0).getTextContent().equals(uuidVC) ||
```

Comment:
```suggestion
            if (node.getAttributes().item(0).getTextContent().equals(uuidVC) ||
```
node.getAttributes() může vratit prazdný list, a .item(0) vyhodí ArrayOutOfBoxException
Mimo jiné, musiš vybrat attribut "resources", který nemusí byt první.